### ISSUE TITLE
Add support for yarn injectEnvironmentFiles

### ISSUE THREAD

**Issue by andrew-w-ross** [wallabyjs/public#3373]
### Issue description or question

Add support for [yarn injectEnvironmentFiles](https://yarnpkg.com/configuration/yarnrc#injectEnvironmentFiles) option. Currently it doesn't respect that setting.  I've found a workaround for now from [this comment](https://github.com/wallabyjs/public/issues/2351#issuecomment-785029824) which works as expected. If anyone else is reading this you'll have to mark the file as executable, `chmod +x .vscode/node.sh`, or this won't work.

### Suggested Solutions

When wallaby detects it's inside a yarn project it should just use `yarn node` as it's runner instead of node.
I'm a little confused as to why this wasn't the general solution when adding support for [yarn2](https://github.com/wallabyjs/public/issues/2351) in the first place. If you did then any changes yarn adds in the future should have just have worked but I might just be naive on the inner workings wallaby. I can report that this works as expected on a pretty large project so it seems like a valid and easy to implement solution.

Alternatively allow setting `wallaby.node` to a command with parameters like `yarn node` as suggested in the workaround comment I linked.

**Comment by smcenlly** [comment]
Our products are integrated with `yarn` using their recommended/documented [`Plug'n'Play` approach](https://yarnpkg.com/features/pnp). 

While the `yarn node` command may work for your case, it's certainly fragile (depends on path variables, etc.) and it does not work universally, specifically for certain scenarios running within a mono-repo / nested yarn configurations.

I'm wondering if this should be addressed by `yarn` when their `.pnp.cjs` and `.pnp.js` files are loaded. I've raised an issue in their repo to ask: https://github.com/yarnpkg/berry/issues/6314.

**Comment by andrew-w-ross** [comment]
@smcenlly  I'm actually running the [nodeLinker with `node-modules`](https://yarnpkg.com/configuration/yarnrc#nodeLinker) on a monorepo with es modules. So on that very specific setup I can confirm it works. I get it you have to deal with the Cartesian product of the node version, package manager, module setup, testing framework and so on. 

I had a feeling that would be the case that's why I suggested the second solution of allowing the `wallaby.node` setting to take a command with parameters.

**Comment by smcenlly** [comment]
> that's why I suggested the second solution of allowing the `wallaby.node` setting to take a command with parameters

While we'd prefer to have this just work by importing the `.pnp.cjs` for the yarn case, there was no reason that multiple command-line arguments can't work as you suggested.

We've updated Wallaby for VS Code to allow you to provide multiple arguments (e.g. `yarn node`) via the `wallaby.node` setting.

**Comment by andrew-w-ross** [comment]
Thanks so much, that worked without a hitch.

> While we'd prefer to have this just work by importing the .pnp.cjs

I get wanting that to be as seamless as possible and for 99.9% of use cases you've succeeded. I first purchased wallaby back in 2018 I had to make sure my `wallaby.js` file was up to date whenever a testing config changed.  With automatic configuration being so polished I don't even remember how to write one. With that being said until wallaby can run off of a protocol like V8 Inspector you're going to have to make some concessions. 

Anyway thanks again I appreciate the quick turn around.
